### ISSUE TITLE
[1.0.0] Unable to use the `vector_math` library

### ISSUE THREAD

**Issue by wolfenrain** [flame-engine/flame#507]
## Description

I am using the `vector_math` library for `Vector3` and other things. The problem is now that I have to pass a `Vector2` value that I receive from the `Game.resize` method to a class that uses the `Vector2` from my own imported `vector_math` which throws the following error:
```
lib/main.dart:50:19: Error: A value of type 'Vector2/*1*/' can't be assigned to a variable of type 'Vector2/*2*/'.
```

## Development enviroment

Flame version:
`1.0.0-rc1`

Flutter doctor:

```
[✓] Flutter (Channel stable, 1.22.2, on Linux, locale en_US.UTF-8)
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/linux#android-setup for more details.
[!] Android Studio (not installed)
[✓] VS Code (version 1.48.2)
[✓] Connected device (1 available)

! Doctor found issues in 2 categories.
```

## Minimal reproducible code (Required for bugs)
```dart
// In test.dart
import 'package:vector_math/vector_math.dart';

class Test {
  Vector2 size;
  
  Test(this.size);
}

// In main.dart
import 'package:flame/extensions/vector2.dart';
import 'package:flutter/material.dart';
import 'package:flame/game/game.dart';
import 'test.dart'

void main() => runApp(MyGame().widget);

class MyGame extends Game {
  @override
  void resize(Vector2 size) {
    var test = Test(size);  
  }

  @override
  void render(Canvas canvas) {}

  @override
  void update(double t) {}
}

```

**Comment by erickzanardo-creditas** [comment]
Isn't just the import of the extension missing?

**Comment by erickzanardo** [comment]
Isn't just the import of the extension missing?

**Comment by wolfenrain** [comment]
Hmm interesting, no the problem was because i was importing `vector_math` instead of `vector_math_64`.. 

But might it be a suggestion that we export the whole `vector_math` easily? So people first dont have to add their own dependency, but also so these kind of mistakes can be prevented?

**Comment by erickzanardo** [comment]
Seems interesting to me! @spydon what are your thoughts ?

**Comment by spydon** [comment]
Sure, I guess with the tree shaking it doesn't matter if we export the whole thing.

**Comment by wolfenrain** [comment]
I would suggest hiding things like `Color` and `Colors` from the `vector_math` as they are not the same as the one we need for `Paint.color`

**Comment by spydon** [comment]
@wolfenrain please review #508

**Comment by spydon** [comment]
Fixed in #508
